src/transformers/trainer.py

```diff
@@ -1158,6 +1158,27 @@ def call_model_init(self, trial=None):
 
         return model
 
+    def torch_jit_model(self, model, training=False, dataloader=None):
+        if not training:
+            jit_inputs = []
+            for example_batch in dataloader:
+                break
+            for key in example_batch:
+                example_tensor = torch.ones_like(example_batch[key])
+                jit_inputs.append(example_tensor)
+            jit_inputs = tuple(jit_inputs)
+            try:
+                jit_model = model.eval()
+                with ContextManagers([self.autocast_smart_context_manager(), torch.no_grad()]):
+                    jit_model = torch.jit.trace(jit_model, jit_inputs, strict=False)
+                jit_model = torch.jit.freeze(jit_model)
+                jit_model(**example_batch)
+                model = jit_model
+            except (RuntimeError, TypeError):
+                logger.warning("fail to use PyTorch jit mode")
```

Comment:
Maybe include the error message in the warning here?

Reply:
have made changes accordingly, thanks.